src/yb/master/ts_descriptor.h

```diff
@@ -177,6 +177,26 @@ class TSDescriptor {
     return leader_count_;
   }
 
+  void set_physical_time(int64_t physical_time) {
+    std::lock_guard<decltype(lock_)> l(lock_);
+    physical_time_ = physical_time;
+  }
+
+  int64_t physical_time() const {
+    SharedLock<decltype(lock_)> l(lock_);
+    return physical_time_;
+  }
+
+  void set_hybrid_time(int64_t hybrid_time) {
+    std::lock_guard<decltype(lock_)> l(lock_);
+    hybrid_time_ = hybrid_time;
+  }
+
+  int64_t hybrid_time() const {
+    SharedLock<decltype(lock_)> l(lock_);
+    return hybrid_time_;
+  }
+
```

Comment:
With this approach, we could render info from different heartbeats in a single view.
Would prefer to add the `TSDescriptorState` struct and update/read it with a single call.
Something like:
```
TSDescriptorState state = {
  .num_live_replicas = req->num_live_tablets(),
  .leader_count = req->leader_count(),
  .physical_time = req->ts_physical_time(),
  .hybrid_time = HybridTime(req->ts_hybrid_time()),
};
desc->set_state(state);
```

Reply:
I have not incorporated this change, since it seems to be a pre-existing issue. Many TSDescriptor fields are set using individual methods, and a larger refactoring may be needed. Let me know if that is okay. Thanks.

Reply:
Yes, it is ok, since we already have this issue. But would be nice to fix in future.